crates/miden-lib/src/account/interface/account_component_interface.rs

```diff
@@ -0,0 +1,249 @@
+                component_interface_vec.push(AccountComponentInterface::Custom(proc_vec));
+            }
+        }
+
+        component_interface_vec
+    }
+
+    /// Generates a body for the note creation of the `send_note` transaction script. The resulting
+    /// code could use different procedures for note creation, which depends on the used interface.
+    ///
+    /// The body consists of two sections:
+    /// - Pushing the note information on the stack.
+    /// - Creating a note:
+    ///   - For basic fungible faucet: pushing the amount of assets and distributing them.
+    ///   - For basic wallet: creating a note, pushing the assets on the stack and moving them to
+    ///     the created note.
+    ///
+    /// # Errors:
+    /// Returns an error if:
+    /// - the interface does not support the generation of the standard `send_note` procedure.
+    /// - the sender of the note isn't the account for which the script is being built.
+    /// - the note created by the faucet doesn't contain exactly one asset.
+    /// - a faucet tries to distribute an asset with a different faucet ID.
+    ///
+    /// # Examples
+    ///
+    /// Example script for the [AccountComponentInterface::BasicWallet] with one note:
+    ///
+    /// ```masm
+    ///     push.{note_information}
+    ///     call.::miden::contracts::wallets::basic::create_note
+    ///
+    ///     push.{note asset}
+    ///     call.::miden::contracts::wallets::basic::move_asset_to_note dropw
+    ///     dropw dropw dropw drop
+    /// ```
+    ///
+    /// Example script for the [AccountComponentInterface::BasicFungibleFaucet] with one note:
+    ///
+    /// ```masm
+    ///     push.{note information}
+    ///     
+    ///     push.{asset amount}
+    ///     call.::miden::contracts::faucets::basic_fungible::distribute dropw dropw drop
+    /// ```
+    pub(crate) fn send_note_body(
+        &self,
+        sender_account_id: AccountId,
+        notes: &[PartialNote],
+    ) -> Result<String, AccountInterfaceError> {
+        let mut body = String::new();
+
+        for partial_note in notes {
+            if partial_note.metadata().sender() != sender_account_id {
+                return Err(AccountInterfaceError::InvalidSenderAccount(
+                    partial_note.metadata().sender(),
+                ));
+            }
+
+            body.push_str(&format!(
+                "push.{recipient}
+                push.{execution_hint}
+                push.{note_type}
+                push.{aux}
+                push.{tag}\n",
+                recipient = word_to_masm_push_string(&partial_note.recipient_digest()),
+                note_type = Felt::from(partial_note.metadata().note_type()),
+                execution_hint = Felt::from(partial_note.metadata().execution_hint()),
+                aux = partial_note.metadata().aux(),
+                tag = Felt::from(partial_note.metadata().tag()),
+            ));
```

Comment:
Here, and below as well, I'd probably add comments with the expected state of the stack after this expression. For example:
```rust
// [tag, aux, note_type, execution_hint, recipient]
```